docs/src/main/paradox/project/project-index.md

```diff
@@ -8,16 +9,16 @@
 * [scala3](scala3.md)
 * [downstream-upgrade-strategy](downstream-upgrade-strategy.md)
 * [../common/may-change](../common/may-change.md)
-* [ide] (../additional/ide.md)
-* [lombok](immutable.md)
-* [osgi](../additional/osgi.md)
+* [../additional/ide](../additional/ide.md)
+* [immutable](immutable.md)
+* [../additional/osgi](../additional/osgi.md)
 * [migration-guides](migration-guides.md)
 * [rolling-update](rolling-update.md)
 * [issue-tracking](issue-tracking.md)
 * [licenses](licenses.md)
-* [license-report](license-report.md) 
-* [faq](../additional/faq.md)
-* [books](../additional/books.md)
+* [license-report](license-report.md)
+* [../additional/faq](../additional/faq.md)
+* [../additional/books](../additional/books.md)
```

Comment:
seems this can be split to another pr.

Reply:
I have rolled back these lines, and to avoid any misunderstanding I provide some redundant explanations here:

- The purpose of changing these lines was to allow ProjectIndexGenerator to match the text and skip writing to this file (when license report is enabled). After I complete the rollback commit, this file will always perform write operations.
- ProjectIndexGenerator cannot be completed in a separate PR because when LicenseReport is disabled, the link in the index file will complain lack of a file, resulting in a compilation failure.